podcasts/Onboarding/Plus/UpgradeExperiment/PlusPaywallContainer.swift

```diff
@@ -0,0 +1,126 @@
+import SwiftUI
+import PocketCastsUtils
+
+struct PlusPaywallContainer: View {
+    @ObservedObject var viewModel: PlusLandingViewModel
+
+    @State var presentSubscriptionView = false
+
+    private let type: ContainerType
+    private let subscriptionInfo: PlusPricingInfoModel.PlusProductPricingInfo?
+    private let tier = UpgradeTier.plus
+
+    private var hasBottomSafeArea: Bool {
+        !UIDevice.current.isiPad() && safeAreaBottomHeight > 0
+    }
+
+    private var safeAreaBottomHeight: CGFloat {
+        (SceneHelper.connectedScene()?.windows.first(where: \.isKeyWindow)?.safeAreaInsets.bottom ?? 0)
+    }
+
+    private var topBar: some View {
+        HStack(spacing: 0) {
+            Spacer()
+            Button(viewModel.source == .upsell ? L10n.eoyNotNow : L10n.plusSkip) {
+                viewModel.dismissTapped()
+            }
+            .foregroundColor(.white)
+            .font(style: .body, weight: .medium)
+            .padding()
+        }
+    }
+
+    private var footer: some View {
+        VStack(spacing: 0) {
+            Button(action: {
+                presentSubscriptionView.toggle()
+            }, label: {
+                Text(L10n.upgradeExperimentPaywallButton)
+            })
+            .buttonStyle(PlusOpaqueButtonStyle(isLoading: false, plan: .plus))
+            .padding(.horizontal, Constants.buttonHPadding)
+
+            if let offer = subscriptionInfo?.offer {
+                Text(offer.experimentDescription)
+                    .font(size: Constants.offerTextSize, style: .body)
+                    .multilineTextAlignment(.center)
+                    .foregroundStyle(.white)
+                    .padding(.top, Constants.offerTextTopPadding)
+            }
+        }
+    }
+
+    init(viewModel: PlusLandingViewModel, type: ContainerType) {
+        self.viewModel = viewModel
+        self.type = type
+
+        let displayProduct = [viewModel.continuePurchasing, viewModel.initialProduct].compactMap { $0 }.first
```

Comment:
@SergioEstevao can I just use `viewModel.initialProduct` as the experiment won't include Patron?

Reply:
I think that the `continuePurchasing` is used to also reflect previous choices in terms of yearly/monthly so if you screen will allow selection between those you should keep it.